### ISSUE TITLE
Confidentiality and wording issues in sharing notification e-mails (when sharing with a circle)

### ISSUE THREAD

**Issue by Jerome-Herbinet** [nextcloud/circles#1239]
Hello,

**When we receive a share from a user, because we belong to a circle, I see two problems.**

**The first problem is in my opinion the most annoying:** 
It says, for example, "_John Doe_ shared multiple files with _Marketing Team_". I think that as the recipient of the share, I don't need to know the entity with which John Doe has shared the file. "_Marketing team_" is in my example something very classic, but the name of the circle may be quite different and have meaning only to John Doe ... and in that case, I may wonder why I am receiving this share. Just imagine also that John Doe names his circle with an "embarrassing" wording hahaha ... you know what I mean?! :-) 
I think this is very important, especially if the circle is private.
So to sum up, putting the name of the circle in the notification email poses a confidentiality/privacy problem. The sentence should simply be "John Doe shared multiple files with you".

**The second problem is that in all cases, it says "multiple files" in the sentence, but this is not possible because a user can only share one file at a time, or one folder at time. So the sentence should be, according to 2 possible cases :** 
- "John Doe shared a file with you"
- "John Doe shared a file with you"

Thanks for reading me.
Jérôme

**Comment by onny** [comment]
It looks like the wording issue has been fixed, can you confirm this?

![image](https://github.com/nextcloud/circles/assets/757752/8340cbe1-62bd-4f2f-8331-98fea8b5c738)

https://github.com/nextcloud/circles/pull/735/files

**Comment by jancborchardt** [comment]
> The sentence should simply be "John Doe shared multiple files with you".

@Jerome-Herbinet wouldn’t this also be confusing as it makes you think they only shared it with you and not others? We should make sure that in the email it is shown the same way as in e.g. the Files sidebar.

**Comment by Jerome-Herbinet** [comment]
> > The sentence should simply be "John Doe shared multiple files with you".
> 
> @Jerome-Herbinet wouldn’t this also be confusing as it makes you think they only shared it with you and not others? We should make sure that in the email it is shown the same way as in e.g. the Files sidebar.

Good remark @jancborchardt ! So the sentence could be something like (depending on what's shared) : 

- "John Doe shared a file with you and some other people."
- "John Doe shared a folder with you and some other people."

PS : As I said in my first message, "multiple files" cannot be used because a user can only share one file or folder at a time.

**Comment by jancborchardt** [comment]
@Jerome-Herbinet how does it currently look like in Files?

"some other people" is a bit strange wording. I would say the least we could say is "[Name] shared [file/folder] with you via a circle". Not sure we need to include "a file" or "a folder" cause we know the name, either we include the name (in the case of the mail) or we leave the generic descriptor out (when it’s in the sidebar).

**Comment by Jerome-Herbinet** [comment]
> @Jerome-Herbinet how does it currently look like in Files?
> 
> "some other people" is a bit strange wording. I would say the least we could say is "[Name] shared [file/folder] with you via a circle". Not sure we need to include "a file" or "a folder" cause we know the name, either we include the name (in the case of the mail) or we leave the generic descriptor out (when it’s in the sidebar).

Why not @jancborchardt ; one question appears : will the majority of "external" people understand what a "circle" is ?

**Comment by Jerome-Herbinet** [comment]
@jancborchardt, now that the “Circles” app has been renamed “Teams”, it might be worth reviving this discussion to see if the attached PR still has a hope of being merged. I asked you a question last time. Do you have an answer ?

**Comment by jancborchardt** [comment]
@Jerome-Herbinet I would say since the rename of Circles to Teams, my suggestion works even better:
> [Name] shared [file/folder] with you via a team

**Comment by Jerome-Herbinet** [comment]
> @Jerome-Herbinet I would say since the rename of Circles to Teams, my suggestion works even better:
> 
> > [Name] shared [file/folder] with you via a team

@jancborchardt “via a team” can lead to misunderstanding or confusion on the part of the message's recipient. He will wonder ... what team are we talking about? I don't even think we should use the term “team”.

I would just say: 

“[Name] shared [file/folder] with you and others.” 

That way, the recipient will know that he's not the only one to benefit from the sharing, but won't waste brain power trying to figure out what this team thing is (especially since, as an external person, he's not at all aware of this team notion, unless the sender of the sharing explains it to him, but that's not very probable). 

Nevertheless, the use of the “team” notion is useful internally in the context of “Activity”, for sharing issuers (but still not for internal and external recipients).

**Comment by jancborchardt** [comment]
I have to say I’m a bit confused, because if you see the notification, you also have to be part of this team. And then the team name definitely is not confidential. So I don’t really see the issue here?

**Comment by Jerome-Herbinet** [comment]
> I have to say I’m a bit confused, because if you see the notification, you also have to be part of this team. And then the team name definitely is not confidential. So I don’t really see the issue here?

OK for "[Name] shared [file/folder] with you via a team"

**Comment by Jerome-Herbinet** [comment]
> via a team

@jancborchardt new commit in linked PR https://github.com/nextcloud/circles/pull/1422

**Comment by jancborchardt** [comment]
> > I have to say I’m a bit confused, because if you see the notification, you also have to be part of this team. And then the team name definitely is not confidential. So I don’t really see the issue here?
> 
> OK for "[Name] shared [file/folder] with you via a team"

@Jerome-Herbinet What regarding my point that you are part of this team and will know the team name if you will get the share? I think the more context, the better here.

**Comment by Jerome-Herbinet** [comment]
> > > I have to say I’m a bit confused, because if you see the notification, you also have to be part of this team. And then the team name definitely is not confidential. So I don’t really see the issue here?
> > 
> > 
> > OK for "[Name] shared [file/folder] with you via a team"
> 
> @Jerome-Herbinet What regarding my point that you are part of this team and will know the team name if you will get the share? I think the more context, the better here.

@jancborchardt I think that there are 2 types of teams members : 
- external members
- internal members (NC instance users)

Messages should be different. 

For external members, the message would be "[Name] shared [file/folder] with you via a team"
For internal members, the message could be approximately the old one, including the team's name

In this case with 2 scenarios, I don't know how to develop the “condition”, based on both of them.

**Comment by jancborchardt** [comment]
> For external members, the message would be "[Name] shared [file/folder] with you via a team"
> For internal members, the message could be approximately the old one, including the team's name

Ah got it, yes that sounds very good to me! :) Sorry for the confusion.

**Comment by Jerome-Herbinet** [comment]
> > For external members, the message would be "[Name] shared [file/folder] with you via a team"
> > For internal members, the message could be approximately the old one, including the team's name
> 
> Ah got it, yes that sounds very good to me! :) Sorry for the confusion.

OK great. As I told you, for that "switch", I'll need someone to co-develop. My scope (skills in this context) only concerns wordings changes.

**Comment by jancborchardt** [comment]
@ArtificialOwl @susnux could you help @Jerome-Herbinet here? :)

**Comment by Jerome-Herbinet** [comment]
Ping @ArtificialOwl
